src/reducers/map.js

```diff
@@ -33,14 +48,24 @@ export default ({
   }
 };
 
-const add = (state, action, keyGetter, itemGetter) => {
-  const key = keyGetter(action);
+const addItem = (state, key, item) => {
   return {
-    byId: {...state.byId, [key]: itemGetter(action)},
+    byId: {...state.byId, [key]: item},
     allIds: state.allIds.includes(key) ? state.allIds : state.allIds.concat(key),
   };
 };
 
+const add = (state, action, keyGetter, itemGetter) => {
+  return addItem(state, keyGetter(action), itemGetter(action));
+};
+
+const addMany = (state, action, addChangeItemsGetter) => {
+  const items = addChangeItemsGetter(action)
+  return Object.keys(items).reduce((result, key) => {
+    return addItem(result, key, items[key]);
```

Comment:
I like your attempt at factorizing some code here. But the performance impact is O(n). Now that addChangeItemsGetter returns the items as a map-Object, you can merge them all at once:
```js
const addMany = (state, action, addChangeItemsGetter) => {
  const items = addChangeItemsGetter(action);
  const keys = Object.keys(items);
  return {
    byId: {...state.byId, ...items},
    allIds: [...new Set(state.allIds.concat(keys))],
  };
}
```

`changeMany` and `removeMany` could use similar optimizations.